openedx/core/djangoapps/api_admin/views.py

```diff
@@ -0,0 +1,86 @@
+"""Views for API management."""
+import logging
+from smtplib import SMTPException
+
+from django.conf import settings
+from django.core.mail import send_mail
+from django.core.urlresolvers import reverse_lazy, reverse
+from django.shortcuts import redirect
+from django.utils.translation import ugettext as _
+from django.views.generic import View
+from django.views.generic.edit import CreateView
+
+from edxmako.shortcuts import render_to_response, render_to_string
+from openedx.core.djangoapps.api_admin.forms import ApiAccessRequestForm
+from openedx.core.djangoapps.api_admin.models import ApiAccessRequest
+
+log = logging.getLogger(__name__)
+
+
+class ApiRequestView(CreateView):
+    """Form view for requesting API access."""
+    form_class = ApiAccessRequestForm
+    template_name = 'api_admin/api_access_request_form.html'
+    success_url = reverse_lazy('api-status')
+
+    def get(self, request):
+        """
+        If the requesting user has already requested API access, redirect
+        them to the client creation page.
+        """
+        if ApiAccessRequest.api_access_status(request.user) is not None:
+            return redirect(reverse('api-status'))
+        return super(ApiRequestView, self).get(request)
+
+    def send_email(self, form):
+        """
+        Send an email to settings.API_ACCESS_MANAGER_EMAIL with the
+        contents of this API access request.
+        """
+        api_request = form.instance
+        company_name = form.cleaned_data['company_name']
+        context = {
+            'approval_url': self.request.build_absolute_uri(
+                reverse('admin:api_admin_apiaccessrequest_change', args=(api_request.id,))
+            ),
+            'company_name': company_name,
```

Comment:
Let's be consistent about populating this context dict.

address is taken directly from the `cleaned_data`, company name is in a variable.  Either is fine, but we should pick one way.  (Personal preference for the way we are doing this for address)